### PR TITLE
Handle Noise Better

### PR DESCRIPTION
**Pull request by rylev**
This adjusts historical analysis to handle noise more quickly. The following changes are included:
* Limit the number of historical runs that are used to compute significance threshold to 30 instead of 100. This means that if a test case starts to become noisy, the significance threshold will more quickly reflect this.
* Expose significance threshold in the compare page. Being able to see the significance threshold will allow readers to more quickly judge whether a test case is historically noisy or not. See below for what this looks like.
* Sort test cases on the compare page by significance factor. The tests cases with the largest change *in comparison to the historical average* will be on top vs. just the test cases with the largest change (which favors noisy test cases).

image

### CODE REVIEW

File: site/static/compare/script.js [hunk 1]
```diff
@@ -359,10 +360,16 @@ app.component('test-cases-table', {
...
-                            href="https://github.com/rust-lang/rustc-perf/blob/master/docs/comparison-analysis.md#what-makes-a-test-result-significant">
-                            here</a> how the significance threshold is calculated.
+                        means the result is 2.5 times over the significance threshold.
+                    </span>
+                </span>
+            </th>
+            <th>
+                Significance Threshold<span class="tooltip">?
+                    <span class="tooltiptext">
+                        The minimum % change that is considered significant. The higher the significance, the noisier a test case is.
```
Comment (Kobzol) [inline site/static/compare/script.js:370]:
```suggestion
                        The minimum % change that is considered significant. The higher the significance threshold, the noisier a test case is.
```

**Review by Kobzol — COMMENTED** [review]
Displaying how noisy a certain benchmark is sounds like a great idea.

I'm not sure about the sorting though. Maybe I'll just need to get used to it. Maybe we could add a checkbox to toggle between sorting by significance/absolute change?

It's probably a bit unintuitive that the tables contain max/min absolute change, but that won't be reflected in the table order.

**Comment by Mark-Simulacrum** [comment]
I think sorting by the change itself still makes sense, rather than the significance factor. Two benchmarks with roughly equivalent significance (e.g., 10x - basically "high") should be ordered by the change each had, which with different noise levels etc might well be different.

**Comment by nnethercote** [comment]
Thanks for this PR, it's good that this will be improving. Comments:
- I think it would be better to have this column order: change, significance threshold, significance factor.
- I agree that sorting by significance factor is surprising, and shouldn't be the default. A toggle for it would be reasonable. Even better would be if we could sort by any of the columns, in both ascending and descending order. (But that's way beyond the scope of this PR.)

**Comment by rylev** [comment]
I've reverted the change to sorting, and I've changed the ordering of the columns to put the threshold first and then the factor.

**Review by Kobzol — APPROVED** [review]

**Pull request merged**